cluster-autoscaler/cloudprovider/aws/aws_cloud_provider_test.go

```diff
@@ -109,6 +111,23 @@ func TestBuildAwsCloudProvider(t *testing.T) {
 	assert.NoError(t, err)
 }
 
+func TestInstanceTypeFallback(t *testing.T) {
+	resourceLimiter := cloudprovider.NewResourceLimiter(
+		map[string]int64{cloudprovider.ResourceNameCores: 1, cloudprovider.ResourceNameMemory: 10000000},
+		map[string]int64{cloudprovider.ResourceNameCores: 10, cloudprovider.ResourceNameMemory: 100000000})
+
+	do := cloudprovider.NodeGroupDiscoveryOptions{}
+	opts := config.AutoscalingOptions{}
+
+	os.Setenv("AWS_REGION", "us-west-2")
+	defer os.Unsetenv("AWS_REGION")
+
+	// This test ensures that no klog.Fatalf calls occur when constructing the AWS cloud provider.  Specifically it is
+	// intended to ensure that instance type fallback works correctly in the event of an error enumerating instance
+	// types.
+	_ = BuildAWS(opts, do, resourceLimiter)
```

Comment:
AFAICT, this test is only testing the happy path, though, right? If we passed in a non-existent AWS region, like "us-south-42", for instance, would that trigger an error being returned from `GenerateEC2InstanceTypes` and therefore trigger the fallback mechanism?

Reply:
Good catch, I tested with no credentials set locally, I assumed there were no credentials in the test environment.  I'll fix it.